File: sherpa-onnx/java-api/src/main/java/com/k2fsa/sherpa/onnx/VersionInfo.java
package com.k2fsa.sherpa.onnx;

public class VersionInfo {

    public static String getVersion() {
        LibraryLoader.maybeLoad();
        return getVersionStr2();
    }

    public static String getGitSha1() {
        LibraryLoader.maybeLoad();
        return getGitSha12();
    }

    public static String getGitDate() {
        LibraryLoader.maybeLoad();
        return getGitDate2();
    }

    private static native String getVersionStr2();
    private static native String getGitSha12();
    private static native String getGitDate2();
}

File: sherpa-onnx/kotlin-api/VersionInfo.kt
package com.k2fsa.sherpa.onnx

class VersionInfo {
    companion object {
        init {
            System.loadLibrary("sherpa-onnx-jni")
        }

        val version: String
            get() = getVersionStr2()

        val gitSha1: String
            get() = getGitSha12()

        val gitDate: String
            get() = getGitDate2()

        external fun getVersionStr2(): String
        external fun getGitSha12(): String
        external fun getGitDate2(): String
    }
}
